src/basic/env-util.c

```diff
@@ -207,6 +211,74 @@ static int env_append(char **r, char ***k, char **a) {
         return 0;
 }
 
+int strv_env_load_with_callback(char ***l, char **fnames, void (*invalid_callback)(const char *file, char ***p, void *userdata), void *userdata) {
+        char **i, **r = *l;
```

Comment:
Ah, you're just moving pre-existing code. In that case that's fine, it should be cleaned up, but it's better to do that in a separate commit.